syntax = "proto3";

package dataanalysis;

service DataAnalysisService {
  rpc Aggregate (AggregateRequest) returns (AggregateResponse);
  rpc Filter (FilterRequest) returns (FilterResponse);
  rpc Transform (TransformRequest) returns (TransformResponse);
}

message AggregateRequest {
  repeated double data = 1;
  string operation = 2; // "sum", "avg", "max", "min"
}

message AggregateResponse {
  double result = 1;
}

message FilterRequest {
  repeated double data = 1;
  string condition = 2; // "greater_than", "less_than", "equal_to"
  double value = 3;
}

message FilterResponse {
  repeated double filtered_data = 1;
}

message TransformRequest {
  repeated double data = 1;
  string operation = 2; // "square", "sqrt", "log"
}

message TransformResponse {
  repeated double transformed_data = 1;
}
